share/sql/alerting.sql

```diff
@@ -2,70 +2,138 @@ SET search_path TO monitoring, public;
 
 BEGIN;
 
-CREATE TYPE check_state_type AS ENUM('OK', 'WARNING', 'CRITICAL', 'UNDEF');
+DROP TYPE IF EXISTS check_state_type CASCADE;
+DROP TYPE IF EXISTS check_results_record CASCADE;
+DROP TABLE IF EXISTS check_results_current;
+DROP TABLE IF EXISTS check_results_history;
+DROP TABLE IF EXISTS check_states;
+DROP TABLE IF EXISTS checks;
 
-CREATE TYPE check_results_record AS (
-	datetime TIMESTAMPTZ,
-	state check_state_type,
-	key VARCHAR(64),
-	value REAL,
-	threshold REAL
-);
+CREATE TYPE check_state_type AS ENUM('OK', 'WARNING', 'CRITICAL', 'UNDEF');
 
 CREATE TABLE checks (
-	check_id SERIAL PRIMARY KEY,
-	host_id INTEGER NOT NULL REFERENCES hosts (host_id),
-	instance_id INTEGER REFERENCES instances (instance_id),
-	enabled BOOLEAN NOT NULL DEFAULT false,
-	name VARCHAR(64) NOT NULL,
-	threshold_w REAL,
-	threshold_c REAL,
-	description TEXT
+  check_id SERIAL PRIMARY KEY,
+  host_id INTEGER NOT NULL REFERENCES hosts (host_id),
+  instance_id INTEGER REFERENCES instances (instance_id),
+  enabled BOOLEAN NOT NULL DEFAULT false,
+  name VARCHAR(64) NOT NULL,
+  threshold_w REAL,
+  threshold_c REAL,
+  description TEXT
 );
 CREATE INDEX idx_checks_host_instance ON checks (host_id, instance_id);
 
 
 CREATE TABLE check_states (
-	check_id INTEGER NOT NULL REFERENCES checks (check_id),
-	key VARCHAR(64),
-	state check_state_type DEFAULT 'UNDEF',
-	PRIMARY KEY (check_id, key)
+  check_id INTEGER NOT NULL REFERENCES checks (check_id),
+  key VARCHAR(64),
+  state check_state_type DEFAULT 'UNDEF',
+  PRIMARY KEY (check_id, key)
+);
+
+CREATE TYPE check_results_record AS (
+  datetime TIMESTAMPTZ,
+  state check_state_type,
+  key VARCHAR(64),
+  value REAL,
+  warning REAL,
+  critical REAL
```

Comment:
I would personnaly make it more obvious that `warning` and `critical` are actually thresholds: `warning_threshold` and `critical_threshold`.

Reply:
I'd like to keep composite type attributes concise.